clouds/clouds-azure/src/main/java/azure/core/AzureArchiveDeployer.java

```diff
@@ -0,0 +1,134 @@
+package azure.core;
+
+
+import azure.core.AzureIdentifiableSunstoneResource.Identification;
+import com.azure.resourcemanager.appservice.models.PublishingProfile;
+import com.azure.resourcemanager.appservice.models.WebApp;
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.wildfly.extras.creaper.commands.deployments.Deploy;
+import org.wildfly.extras.creaper.commands.deployments.Undeploy;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import sunstone.core.TimeoutUtils;
+import sunstone.core.api.SunstoneArchiveDeployer;
+import sunstone.core.exceptions.IllegalArgumentSunstoneException;
+import sunstone.core.exceptions.SunstoneException;
+import sunstone.core.exceptions.UnsupportedSunstoneOperationException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+
+/**
+ * Purpose: handle deploy operation to WildFly.
+ *
+ * Heavily uses {@link AzureIdentifiableSunstoneResource} to determine the destination of deploy operation
+ *
+ * To retrieve Azure cloud resources, the class relies on {@link AzureIdentifiableSunstoneResource#get(Annotation, AzureSunstoneStore, Class)}.
+ *
+ * Undeploy operations are registered in the extension store so that they are closed once the store is closed
+ */
+public class AzureArchiveDeployer implements SunstoneArchiveDeployer {
+
+    static void deployToWebApp(Identification resourceIdentification, InputStream is, AzureSunstoneStore store) throws Exception {
+        Path tempFile = Files.createTempFile("sunstone-war-deployment-", ".war");
+        Files.copy(is, tempFile, StandardCopyOption.REPLACE_EXISTING);
+        WebApp azureWebApp = resourceIdentification.get(store, WebApp.class);
+
+        Exception ex = null;
+        for (int i = 0; i < 5; i++) {
+            try {
+                azureWebApp.warDeploy(tempFile.toFile());
+                ex = null;
+                break;
+            } catch (Exception e) {
+                AzureLogger.DEFAULT.debug("WebApp::warDeploy try no. {} failed", i);
+                e.printStackTrace();
+                ex = e;
+                // try another time
+                Thread.sleep(10 * 1000);
```

Comment:
This is OK?

Reply:
`TimeoutUtils.adjust()` should have been used. But I have got rid it using async operation and waiting for the result.

Reply:
Ok. Now I am happy :)